rotkehlchen/exchanges/bybit.py

```diff
@@ -312,12 +312,23 @@ def query_online_trade_history(
         Since we have a clear limit to stop querying what we do is use the startTime/endTime keys
         to filter the data that we need.
         """
-        new_trades = []
+        new_trades: list[Trade] = []
         upper_ts, lower_ts = end_ts, Timestamp(end_ts - WEEK_IN_SECONDS)
-        two_years_ago = Timestamp(ts_now() - DAY_IN_SECONDS * 365 * 2)
-        if start_ts < two_years_ago:
-            # bybit allows to query up to two years into the past
-            start_ts = two_years_ago
+        if self.is_unified_account is True:
+            # unified account can query up to 2 years into the past
+            query_range_limit_ts = Timestamp(ts_now() - DAY_IN_SECONDS * 365 * 2)
+        else:
+            # classic accounts can query 180 days into the past
+            query_range_limit_ts = Timestamp(ts_now() - DAY_IN_SECONDS * 180)
+
+        if end_ts <= query_range_limit_ts:
+            # if the newest trades ts is already out of the query range exit since we won't
```

Comment:
shouldn't you try to query between start and query_range_limit_ts ?

Reply:
Say that the limit is two years. Then we have

```
       <- newer ...... older ->
TWO_YEARS_AGO ... end_ts ... start_ts  
```

so we can't query anything in `[end_ts, start_ts]`